Admin serch filter jpa repository

GoralTomaszGorski opened the pull request:
Hallo Marcin

last time I worked to add searching patients for name or surname.
last experiences allow me to say that rule KISS is the best rule from others rules as well in programming as army;)

Today I fell stupid because of I wasted my time to tried prepared complex code only for at the finish write simply

  @Repository
List<Patient> findBySurnameContainsIgnoreCaseOrNameContainsIgnoreCase(String surname, String name);

@Service
patientJpaRepository.findBySurnameContainsIgnoreCaseOrNameContainsIgnoreCase(keyword, keyword)
                .stream()
                .map(PatientDtoMapper::map)
                .toList();
    }

GoralTomaszGorski committed:
redundant code meeting service fixed
preparing to search Filter JpaRepository

GoralTomaszGorski committed:
redundant code meeting service fixed
preparing to search Filter JpaRepository

gorskimarcin96 commented on src/main/java/goral/psychotherapistoffice/domain/meeting/MeetingService.java:
```diff
@@ -36,13 +42,17 @@ public MeetingService(PatientRepository patientRepository, CalenderRepository ca
-        Calender calender = calenderRepository.findById(meetingToSaveDto.getCalender()).orElseThrow();
-        meeting.setCalender(calender);
-        calender.setFree(false);
+        Calender calender = calenderRepository.findCalenderByIdAndFreeIsTrue(meetingToSaveDto.getCalender()).orElseThrow();
```
i think we can set name little easier, `findCalenderByIdAndIsFree`. for negation we can set `findCalenderByIdAndIsNotFree`

GoralTomaszGorski replied:
I don't understand, I don't need booked meetings here

I try it but dosn't work
findCalenderByIdAndFree(long) 
expects at least 2 arguments but only found 1;

gorskimarcin96 commented on src/main/java/goral/psychotherapistoffice/domain/meeting/MeetingService.java:
```diff
@@ -55,28 +65,20 @@ public List<MeetingDto> findAllMeetings() {
+        if (calender.isFree()) {
+            meeting.setCalender(calender);
+            calender.setFree(false);
+        } else {
```
is it possible run this condition? if yes you should catch this exception

GoralTomaszGorski replied:
It is possible because I can click a orange icon, link in thymeleaf It can redirect me to this booked calender term

OK I will try to fix it Sunday or Monday Thank you Marcin for your time and help

gorskimarcin96 reviewed (changes requested):
writing code is process. on 14 february i spent 10 hours for creating singletons in javascript and i had a lot changes. that all to trash, because JS is working asynchronous. my design pattern isnt effective for my case. yesterday i found other solution, but again i musted (can i write "musted", probably no?)... but again i had spend few hours

GoralTomaszGorski commented on src/main/java/goral/psychotherapistoffice/domain/meeting/MeetingService.java:
```diff
@@ -55,28 +65,20 @@ public List<MeetingDto> findAllMeetings() {
     }
```
You tough probably  about it

    List<Meeting>findAllByCalenderNotFreeOrderByCalender;
but It give null

GoralTomaszGorski committed:
redundant code meeting service fixed
search Filter for name and surname
addMeeting shorted
addMeetingWithNewPatient ResponseStatusException added
